independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationsTransformationContext.java

```diff
@@ -57,12 +67,29 @@ void setAnnotations(C annotations) {
     }
 
     public Collection<AnnotationInstance> getAllAnnotations() {
+        return getAllAnnotationForTarget(getTarget());
+    }
+
+    public Collection<AnnotationInstance> getAllTargetAnnotations() {
+        return getAllAnnotationForTarget(getAnnotationTarget());
+    }
+
+    private Collection<AnnotationInstance> getAllAnnotationForTarget(AnnotationTarget target) {
         AnnotationStore annotationStore = get(BuildExtension.Key.ANNOTATION_STORE);
         if (annotationStore == null) {
             throw new IllegalStateException(
                     "Attempted to use the getAllAnnotations() method but AnnotationStore wasn't initialized.");
         }
-        return annotationStore.getAnnotations(getTarget());
+        // Jandex overlay in compatible mode won't allow us to query for METHOD_PARAMETER
+        // hence if the "target" is method param, we query whole method and filter it
+        if (AnnotationTarget.Kind.METHOD_PARAMETER.equals(target.kind())) {
+            return annotationStore.getAnnotations(target.asMethodParameter().method()).stream()
```

Comment:
We have quite a few util methods declared on `io.quarkus.arc.processor.Annotations`. Have you tried to use one of them or add a new one so that it could be easily reused?